### PR TITLE
Set device for local PyTorchGemmaLanguageModel

### PR DESCRIPTION
**Pull request by mkielo3**
Small change to allow user to specify a device for local LLMs. There doesn't seem to be a supported way to do this. Currently "pytorch_gemma_local.ipynb" defaults to cpu and runs in 19 seconds vs. 0.5 seconds on GPU. Apologies if I missed something.

### CODE REVIEW

File: concordia/language_model/utils.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def language_model_setup(
     model_name: str,
     api_key: str | None = None,
     disable_language_model: bool = False,
+    device: str = 'cpu'
 ) -> language_model.LanguageModel:
   """Get the wrapped language model.
```
Comment (jagapiou) [inline concordia/language_model/utils.py:38]: Please add the doc for the device argument.

File: concordia/language_model/pytorch_gemma_model.py [hunk 2]
```diff
@@ -46,11 +47,12 @@ def __init__(
     """
```
Comment (jagapiou) [inline concordia/language_model/pytorch_gemma_model.py:47]: Please add the doc for the device argument.

File: concordia/language_model/pytorch_gemma_model.py [hunk 3]
```diff
@@ -35,6 +35,7 @@ def __init__(
       *,
       measurements: measurements_lib.Measurements | None = None,
       channel: str = language_model.DEFAULT_STATS_CHANNEL,
+      device: str = 'cpu'
```
Comment (jagapiou) [inline concordia/language_model/pytorch_gemma_model.py:38]: consider: could this be `Literal['cpu', 'gpu']` or somesuch to make it clearer what values are OK?

File: concordia/language_model/utils.py [hunk 4]
```diff
@@ -32,6 +32,7 @@ def language_model_setup(
     model_name: str,
     api_key: str | None = None,
     disable_language_model: bool = False,
+    device: str = 'cpu'
```
Comment (jagapiou) [inline concordia/language_model/utils.py:35]: consider: could this be `Literal['cpu', 'gpu']` or somesuch to make it clearer what values are OK?

**Review by jagapiou — CHANGES_REQUESTED** [review]
Hi thanks for the changes, looks good.

Just some minor changes please. And then please squash your commit on HEAD.

File: concordia/language_model/utils.py (hunk 1 above)
Comment (mkielo3) [reply concordia/language_model/utils.py:38]: Done, thanks for flagging

File: concordia/language_model/pytorch_gemma_model.py (hunk 2 above)
Comment (mkielo3) [reply concordia/language_model/pytorch_gemma_model.py:48]: Done, thanks for flagging

File: concordia/language_model/pytorch_gemma_model.py (hunk 3 above)
Comment (mkielo3) [reply concordia/language_model/pytorch_gemma_model.py:38]: PyTorch allows user to specify the device with custom strings. Ex: "cuda:0" for GPU 1, "cuda:1" for GPU 2, etc. so there isn't really way to wrap this nicely into a literal

File: concordia/language_model/utils.py (hunk 4 above)
Comment (mkielo3) [reply concordia/language_model/utils.py:35]: PyTorch allows user to specify the device with custom strings. Ex: "cuda:0" for GPU 1, "cuda:1" for GPU 2, etc. so there isn't really way to wrap this nicely into a literal

**Review by jagapiou — APPROVED** [review]

**Comment by jagapiou** [comment]
Thanks! You can ignore that failed internal check, this will go in no probs once internally approved.

**Comment by jagapiou** [comment]
merged in 8a0e489930c73f6c5d24e82bdf4ea81594b36bb1

**Pull request closed without merging**